src/content/studies/Controller.test.tsx

```diff
@@ -0,0 +1,196 @@
+import {
+  GET_APPROVED_STUDY,
+  GetApprovedStudyInput,
+  GetApprovedStudyResp,
+  LIST_APPROVED_STUDIES,
+  ListApprovedStudiesInput,
+  ListApprovedStudiesResp,
+} from "../../graphql";
+
+// NOTE: Omitting fields depended on by the component
+const baseUser: Omit<User, "role"> = {
+  _id: "",
+  firstName: "",
+  lastName: "",
+  userStatus: "Active",
+  IDP: "nih",
+  email: "",
+  organization: null,
+  dataCommons: [],
+  createdAt: "",
+  updateAt: "",
+  studies: null,
+};
+
+type ParentProps = {
+  role: User["role"];
+  initialEntry?: string;
+  mocks?: MockedResponse[];
+  ctxStatus?: AuthContextStatus;
+  children: React.ReactNode;
+};
+
+const TestParent: FC<ParentProps> = ({
+  role,
+  initialEntry = "/studies",
+  mocks = [],
+  ctxStatus = AuthContextStatus.LOADED,
+  children,
+}: ParentProps) => {
+  const baseAuthCtx: AuthContextState = useMemo<AuthContextState>(
+    () => ({
+      status: ctxStatus,
+      isLoggedIn: role !== null,
+      user: { ...baseUser, role },
+    }),
+    [role, ctxStatus]
+  );
+
+  return (
+    <MockedProvider mocks={mocks} showWarnings>
+      <MemoryRouter initialEntries={[initialEntry]}>
+        <Routes>
+          <Route
+            path="/studies/:studyId?"
+            element={
+              <AuthContext.Provider value={baseAuthCtx}>
+                <SearchParamsProvider>{children}</SearchParamsProvider>
+              </AuthContext.Provider>
+            }
+          />
+          <Route path="/" element={<div>Root Page</div>} />
+        </Routes>
+      </MemoryRouter>
+    </MockedProvider>
+  );
+};
+
+describe("StudiesController", () => {
+  it("should render the page without crashing", async () => {
+    const listApprovedStudiesMock: MockedResponse<
+      ListApprovedStudiesResp,
+      ListApprovedStudiesInput
+    > = {
+      request: {
+        query: LIST_APPROVED_STUDIES,
+      },
+      variableMatcher: () => true,
+      result: {
+        data: {
+          listApprovedStudies: {
+            total: 1,
+            studies: [
+              {
+                _id: "study-id-1",
+                studyName: "Study Name 1",
+                studyAbbreviation: "SN1",
+                dbGaPID: "db123456",
+                controlledAccess: true,
+                openAccess: false,
+                PI: "Dr. Smith",
+                ORCID: "0000-0001-2345-6789",
+                createdAt: "2022-01-01T00:00:00Z",
+                originalOrg: "",
+              },
+            ],
+          },
+        },
+      },
+    };
+
+    const { getByTestId } = render(
+      <TestParent
+        role="Admin"
+        ctxStatus={AuthContextStatus.LOADED}
+        mocks={[listApprovedStudiesMock]}
+      >
+        <StudiesController />
+      </TestParent>
+    );
+
+    await waitFor(() => {
+      expect(getByTestId("list-studies-container")).toBeInTheDocument();
+    });
+  });
+
+  it("should show a loading spinner when the AuthCtx is loading", async () => {
```

Comment:
I think the reason this is passing is because of the loader within the Manage Studies table. It doesn't appear that the controller checks for the AuthContext status.

I could be wrong though, I just logged the content of the test container and it looks like it's the table that's in the document.

Reply:
I totally agree with you and meant to add this spinner, just forgot. I updated the spinner props so that we can pass a custom test id to make sure we are testing for the right spinner. Updated in [b9aa91d](https://github.com/CBIIT/crdc-datahub-ui/pull/469/commits/b9aa91d29c4be1174e90d1ccc4b6e267d780940c), [92ecc19](https://github.com/CBIIT/crdc-datahub-ui/pull/469/commits/92ecc194f5bfe61657b0ab6c32a71825133ff479), and [a56f9ae](https://github.com/CBIIT/crdc-datahub-ui/pull/469/commits/a56f9ae0885ffdfb10c70e5043dc85e04f6c4542).